### DISCUSSION TITLE
Superface.ai traffic junction problem

### DISCUSSION THREAD

**Discussion by rydlobeton** [superfaceai/.github#9] [Q&A] [answered]
Hello everybody using Superface (always using latest sdk).

I am using Superface for my api connections projects since early 2022 and I am satisfied with it capabilitites. 

Unfortunatelly in my latest project I have found a problem I cannot solve myself. 

When I init and use more than one Profile in my script (nodeJS) it use always one provider, although I set another one.

Example:

provider1 = await sdk.getProvider("provider1");
provider2 = await sdk.getProvider("provider2");
provider3 = await sdk.getProvider("provider3");

profile = await sdk.getProfile("example");
profile2 = await sdk.getProfile("example2");

.getUseCase("example").perform({data}, {provider3}); // uses provider3 as written
.getUseCase("example2").perform({data},{provider2});  // uses provider1 number one although both are set up in super.json and no priorities are written.

If I use in JS script just:

provider2 = await sdk.getProvider("provider2");
.getUseCase("example2").perform({data},{provider2});

it works as intended.

Do you have any idea what can cause the problem?

I would be glad for any help.

Thank you very much in advance!

Rydlobeton

**Comment by martyndavies** [comment]
Hey @rydlobeton, thanks for your question. We've flagged this up with the engineering team, and we'll get you a response ASAP.

**Reply by freaz** [reply, marked as answer]
Hello, from the samples you sent, it looks like you are not passing the provider option correctly. It needs to be set to `provider` property. Like this:

```js
const provider1 = await sdk.getProvider('provider1');
const provider2 = await sdk.getProvider('provider2');
const provider3 = await sdk.getProvider('provider3');

const profile = await sdk.getProfile('example');
const profile2 = await sdk.getProfile('example2');

const res1 = await profile.getUseCase('example').perform({ data }, { provider: provider3 });
const res2 = await profile2.getUseCase('example2').perform({ data }, { provider: provider2 });
```

Or there is a shorter way, by passing provider name as a string:

```js
const profile = await sdk.getProfile('example');
const profile2 = await sdk.getProfile('example2');

const res1 = await profile.getUseCase('example').perform({ data }, { provider: 'provider3' });
const res2 = await profile2.getUseCase('example2').perform({ data }, { provider: 'provider2' });
```

I hope, it resolves your issue,
Ondrej

**Reply by rydlobeton** [reply]
Hello, 

yes it solved the issue. Thank you very much!

On your doc [webpage](https://superface.ai/docs/guides/using-multiple-providers) had written only this example:

`const client = new OneClient();

async function run() {
  const profile = await client.getProfile('email-communication/email-sending');

  const result = await profile
    .getUseCase('SendEmail')
    .perform(inputs, provider);

  console.log(result.unwrap());
}`

It would be nice to see all the possibilities.

Nice job with Superface anyway, it saves me a lot of time!

Best regards,

Rydlobeton
